firstvoices/backend/search/tasks/site_content_indexing_tasks.py

```diff
@@ -11,25 +12,26 @@
     VideoDocumentManager,
 )
 from backend.tasks.utils import ASYNC_TASK_END_TEMPLATE, ASYNC_TASK_START_TEMPLATE
-
-# special tasks for bulk Site content indexing actions
+from firstvoices.celery import link_error_handler
 
 
+# special tasks for bulk Site content indexing actions
 def remove_all(document_manager, queryset):
-    for instance in queryset:
-        document_manager.remove_from_index(instance.id)
+    for instance_id in queryset:
+        document_manager.remove_from_index(instance_id)
 
 
 @shared_task
-def remove_all_site_content_from_indexes(site_id):
-    site = Site.objects.get(id=site_id)
+def remove_all_site_content_from_indexes(site_title, site_content_ids):
+    logger = get_task_logger(__name__)
+    logger.info(ASYNC_TASK_START_TEMPLATE, f"site: {site_title}")
 
-    remove_all(DictionaryEntryDocumentManager, site.dictionaryentry_set.all())
-    remove_all(SongDocumentManager, site.song_set.all())
-    remove_all(StoryDocumentManager, site.story_set.all())
-    remove_all(AudioDocumentManager, site.audio_set.all())
-    remove_all(ImageDocumentManager, site.image_set.all())
-    remove_all(VideoDocumentManager, site.video_set.all())
+    remove_all(DictionaryEntryDocumentManager, site_content_ids["dictionaryentry_set"])
```

Comment:
This needs to take the site id as the parameter and look up the rest at runtime, like it was before. This is in case the content has changed in between the task being queued and starting to run

Reply:
Oh wait, I bet all the content has been deleted by then, right? Was this a fix for that?